log/sentry.go

```diff
@@ -71,12 +74,32 @@ func InitSentry(cfg SentryConfig) (io.Closer, error) {
 	if cfg.SampleRate != nil && *cfg.SampleRate >= 0 && *cfg.SampleRate <= 1 {
 		sampleRate = *cfg.SampleRate
 	}
+
+	// ensure first stack trace frame that is part of this package will be
+	// marked as not in-app to improve the error display in Sentry
+	beforeSend := func(event *sentry.Event, hint *sentry.EventHint) *sentry.Event {
+		for i, exception := range event.Exception {
+			for j := len(exception.Stacktrace.Frames) - 1; j > 0; j-- {
+				frame := exception.Stacktrace.Frames[j]
+				if frame.Function == "ErrorWithSentry" && frame.Module == "github.com/reddit/baseplate.go/log" {
```

Comment:
We also have `ErrorWithSentryWrapper`, which has [a lambda in it](https://github.com/reddit/baseplate.go/blob/v0.8.3/log/wrapper.go#L205), which won't satisfy this loop.

Maybe we should remove the function name check and only check for module?

Reply:
Great point. I think it should be okay to remove it, alternatively we could do
```go
strings.HasPrefix(frame.Function, "ErrorWithSentry")
```
which captures:
```
ErrorWithSentry
ErrorWithSentryWrapper.func1
```

The only reason to include a check on the function name would be to ensure we are not missing other Sentry errors that originate from the baseplate package. But then again… we will still have the information in Sentry, so not a big deal.

Reply:
If nothing is lost, just marked differently on the ui, I'd prefer to do prefix check on the module to mark everything under `github.com/reddit/baseplate.go` as not in-app, over only excluding log package.

Reply:
maybe we want to do:

```
const base = "github.com/reddit/baseplate.go"
if frame.Module == base || strings.HasPrefix(frame.Module, base + "/") {
...
```

in case we are gonna do `github.com/reddit/baseplate.go-upgrader` in the future :)

Reply:
Agreed, this should give us the best default behavior and catch all cases. Also updated the comment.

Reply:
@fishy wouldn't we want to mark something like `github.com/reddit/baseplate.go-upgrader` as not in-app too? Though generally we probably shouldn't mark as not in-app outside of the domain of this package.

Reply:
no `github.com/reddit/baseplate.go-upgrader` could be an app in itself, and things under that prefix should be in-app.

but this is just a very hypothetical scenario and I don't feel strongly either way.